CliFx/Services/CommandInitializer.cs

```diff
@@ -10,67 +12,71 @@ namespace CliFx.Services
     /// </summary>
     public class CommandInitializer : ICommandInitializer
     {
-        private readonly ICommandOptionInputConverter _commandOptionInputConverter;
+        private readonly ICommandInputConverter _commandInputConverter;
         private readonly IEnvironmentVariablesParser _environmentVariablesParser;
 
         /// <summary>
         /// Initializes an instance of <see cref="CommandInitializer"/>.
         /// </summary>
-        public CommandInitializer(ICommandOptionInputConverter commandOptionInputConverter, IEnvironmentVariablesParser environmentVariablesParser)
+        public CommandInitializer(ICommandInputConverter commandInputConverter, IEnvironmentVariablesParser environmentVariablesParser)
         {
-            _commandOptionInputConverter = commandOptionInputConverter;
+            _commandInputConverter = commandInputConverter;
             _environmentVariablesParser = environmentVariablesParser;
         }
 
         /// <summary>
         /// Initializes an instance of <see cref="CommandInitializer"/>.
         /// </summary>
         public CommandInitializer(IEnvironmentVariablesParser environmentVariablesParser)
-            : this(new CommandOptionInputConverter(), environmentVariablesParser)
+            : this(new CommandInputConverter(), environmentVariablesParser)
         {
         }
 
         /// <summary>
         /// Initializes an instance of <see cref="CommandInitializer"/>.
         /// </summary>
         public CommandInitializer()
-            : this(new CommandOptionInputConverter(), new EnvironmentVariablesParser())
+            : this(new CommandInputConverter(), new EnvironmentVariablesParser())
         {
         }
 
-        /// <inheritdoc />
-        public void InitializeCommand(ICommand command, CommandSchema commandSchema, CommandInput commandInput)
+        private void InitializeCommandOptions(ICommand command, TargetCommandSchema targetCommandSchema)
         {
+            if (targetCommandSchema.Schema is null)
+            {
+                throw new ArgumentException("Cannot initialize command without a schema.");
+            }
+
```

Comment:
Can this actually happen?

Reply:
No. `targetCommandSchema.Schema` is nullable, so this is to avoid the warning. Another way around that would be the `!` operater when dereferencing, but I feel this approach is less prone to errors and more clearly states "I expect this property to not be null when we enter this method".

Reply:
Ok I see, that makes sense. In which cases can `targetCommandSchema.Schema` be `null`? Can we avoid it somehow, making it non-nullable?

Reply:
It probably could, I will look into it.